File: extensions/spark/kyuubi-spark-connector-tpcds/src/main/java/io/trino/tpcds/row/HouseholdDemographicsRow.java
package io.trino.tpcds.row;

    import java.util.List;

    import org.apache.kyuubi.spark.connector.tpcds.row.KyuubiTPCDSTableRowWithNulls;

    import static com.google.common.collect.Lists.newArrayList;
    import static io.trino.tpcds.generator.HouseholdDemographicsGeneratorColumn.HD_BUY_POTENTIAL;
    import static io.trino.tpcds.generator.HouseholdDemographicsGeneratorColumn.HD_DEMO_SK;
    import static io.trino.tpcds.generator.HouseholdDemographicsGeneratorColumn.HD_DEP_COUNT;
    import static io.trino.tpcds.generator.HouseholdDemographicsGeneratorColumn.HD_INCOME_BAND_ID;
    import static io.trino.tpcds.generator.HouseholdDemographicsGeneratorColumn.HD_VEHICLE_COUNT;

public class HouseholdDemographicsRow
    extends KyuubiTPCDSTableRowWithNulls
{
  private final long hdDemoSk;
  private final long hdIncomeBandId;
  private final String hdBuyPotential;
  private final int hdDepCount;
  private final int hdVehicleCount;

  public HouseholdDemographicsRow(long nullBitMap, long hdDemoSk, long hdIncomeBandId, String hdBuyPotential, int hdDepCount, int hdVehicleCount)
  {
    super(nullBitMap, HD_DEMO_SK);
    this.hdDemoSk = hdDemoSk;
    this.hdIncomeBandId = hdIncomeBandId;
    this.hdBuyPotential = hdBuyPotential;
    this.hdDepCount = hdDepCount;
    this.hdVehicleCount = hdVehicleCount;
  }

  public long getHdDemoSk() {
    return hdDemoSk;
  }

  public long getHdIncomeBandId() {
    return hdIncomeBandId;
  }

  public String getHdBuyPotential() {
    return hdBuyPotential;
  }

  public int getHdDepCount() {
    return hdDepCount;
  }

  public int getHdVehicleCount() {
    return hdVehicleCount;
  }

  @Override public Object[] values() {
    return new Object[] {
        getOrNullForKey(hdDemoSk, HD_DEMO_SK),
        getOrNullForKey(hdIncomeBandId, HD_INCOME_BAND_ID),
        getOrNull(hdBuyPotential, HD_BUY_POTENTIAL),
        getOrNull(hdDepCount, HD_DEP_COUNT),
        getOrNull(hdVehicleCount, HD_VEHICLE_COUNT)
    };
  }
}

File: extensions/spark/kyuubi-spark-connector-tpcds/src/main/scala/io/trino/tpcds/row/HouseholdDemographicsRow.scala
/*
 * Licensed to the Apache Software Foundation (ASF) under one or more
 * contributor license agreements.  See the NOTICE file distributed with
 * this work for additional information regarding copyright ownership.
 * The ASF licenses this file to You under the Apache License, Version 2.0
 * (the "License"); you may not use this file except in compliance with
 * the License.  You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package io.trino.tpcds.row

import java.util.{List => JList}

import io.trino.tpcds.generator.HouseholdDemographicsGeneratorColumn._
import org.apache.spark.sql.catalyst.InternalRow
import org.apache.spark.sql.catalyst.expressions.GenericInternalRow

class HouseholdDemographicsRow(
    nullBitMap: Long,
    private val hdDemoSk: Long,
    private val hdIncomeBandId: Long,
    private val hdBuyPotential: String,
    private val hdDepCount: Int,
    private val hdVehicleCount: Int) extends TableRowWithNulls(nullBitMap, HD_DEMO_SK)
  with KyuubiTableRowWithNulls {

  override val nullBitMapInternal = nullBitMap
  override val firstColumnInternal = HD_DEMO_SK

  def getHdDemoSk: Long = hdDemoSk
  def getHdIncomeBandId: Long = hdIncomeBandId
  def getHdBuyPotential: String = hdBuyPotential
  def getHdDepCount: Int = hdDepCount
  def getHdVehicleCount: Int = hdVehicleCount

  override def getValues: JList[String] = throw new UnsupportedOperationException()

  def internalRow: InternalRow =
    new GenericInternalRow(
      Array(
        getLongOrNull(hdDemoSk, HD_DEMO_SK),
        getLongOrNull(hdIncomeBandId, HD_INCOME_BAND_ID),
        getStringOrNullInternal(hdBuyPotential, HD_BUY_POTENTIAL),
        getIntOrNull(hdDepCount, HD_DEP_COUNT),
        getIntOrNull(hdVehicleCount, HD_VEHICLE_COUNT)))
}
